app/controllers/campaign_pages_controller.rb

```diff
@@ -25,6 +25,7 @@ def new
 
   def create
     @campaign_page = CampaignPage.new(@page_params)
+    @campaign_page.compile_html
```

Comment:
I removed `compile_html` as a `before_save` because it was causing broken tests that don't happen with this way of doing things. This adds an extra line to the controller but gives us more control over when and where we compile the HTML, as I could see things getting thorny (if we compile HTML when we first save a page, but the widgets aren't saved until after the page is, the compiled page would be inaccurate - this gives us greater control). Open to thoughts on whether this is the right way to handle it.